From 90e11d3713f68fece0b28cc2a1bda3dde8aa3303 Mon Sep 17 00:00:00 2001
From: Christian Dywan <christian@twotoasts.de>
Date: Tue, 1 Feb 2011 01:52:39 +0100
Subject: [PATCH] Only no adblock if main frame in provisional state

---
 extensions/adblock.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/extensions/adblock.c b/extensions/adblock.c
index 6ae7adc..ff08b67 100644
--- a/extensions/adblock.c
+++ b/extensions/adblock.c
@@ -753,7 +753,8 @@ adblock_resource_request_starting_cb (WebKitWebView*         web_view,
     const char *page_uri;
 
     /* Never filter the main page itself */
-    if (web_frame == webkit_web_view_get_main_frame (web_view))
+    if (web_frame == webkit_web_view_get_main_frame (web_view)
+     && webkit_web_frame_get_load_status (web_frame) == WEBKIT_LOAD_PROVISIONAL)
         return;
 
     req_uri = webkit_network_request_get_uri (request);
-- 
1.7.2.3

