From a205df3970659d0956c21641d251824da2d7d2b3 Mon Sep 17 00:00:00 2001
From: Christian Dywan <christian@twotoasts.de>
Date: Sun, 20 Feb 2011 20:07:29 +0100
Subject: [PATCH] Don't limit search engines shown in completion popup

---
 midori/midori-locationaction.c |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/midori/midori-locationaction.c b/midori/midori-locationaction.c
index 000fec2..d54ac43 100644
--- a/midori/midori-locationaction.c
+++ b/midori/midori-locationaction.c
@@ -505,8 +505,6 @@ midori_location_action_popup_timeout_cb (gpointer data)
             g_free (uri);
             g_free (title);
             i++;
-            if (i > 4)
-                break;
         }
         searches += i;
     }
-- 
1.7.4.1

