From b81370bd81f782836bbabdcca4cf7cb219ed53cf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andr=C3=A9=20St=C3=B6sel?= <andre@stoesel.de>
Date: Mon, 5 Sep 2011 18:02:42 +0200
Subject: [PATCH] Fix for Bug #819688

---
 midori/midori-panel.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/midori/midori-panel.c b/midori/midori-panel.c
index a7abc81..5d6dad0 100644
--- a/midori/midori-panel.c
+++ b/midori/midori-panel.c
@@ -13,6 +13,8 @@
 
 #include "midori-view.h"
 
+#include "midori-browser.h"
+
 #include "marshal.h"
 #include "sokoke.h"
 
@@ -588,6 +590,7 @@ static void
 midori_panel_viewable_destroy_cb (GtkWidget*   viewable,
                                   MidoriPanel* panel)
 {
+    MidoriBrowser* browser = midori_browser_get_for_widget ( GTK_WIDGET (panel));
     gint n_pages;
     gchar* action_name;
     GtkAction* action;
@@ -601,7 +604,7 @@ midori_panel_viewable_destroy_cb (GtkWidget*   viewable,
         viewable, midori_panel_viewable_destroy_cb, panel);
 
     n_pages = midori_panel_get_n_pages (panel);
-    if (n_pages > 0)
+    if (n_pages > 0 && browser && !g_object_get_data (G_OBJECT (browser), "midori-browser-destroyed"))
         midori_panel_set_current_page (panel, (n_pages-1 > i) ? i : n_pages - 1);
 
     action_name = g_strconcat ("PanelPage",
-- 
1.7.6.1

