Index: midori-debian-repo/midori/midori-session.c
===================================================================
--- midori-debian-repo.orig/midori/midori-session.c
+++ midori-debian-repo/midori/midori-session.c
@@ -189,7 +189,7 @@ midori_load_soup_session (gpointer setti
     #ifndef G_OS_WIN32
     g_object_set (session,
                   "ssl-use-system-ca-file", TRUE,
-                  "ssl-strict", FALSE,
+                  "ssl-strict", TRUE,
                   NULL);
     #else /* G_OS_WIN32 */
     /* We cannot use "ssl-use-system-ca-file" on Windows
