Index: midori/midori/midori-session.c
===================================================================
--- midori.orig/midori/midori-session.c	2013-10-23 20:49:21.000000000 -0700
+++ midori/midori/midori-session.c	2013-10-23 20:49:48.000000000 -0700
@@ -172,7 +172,7 @@
     #if defined (HAVE_LIBSOUP_2_37_1)
     g_object_set (session,
                   "ssl-use-system-ca-file", TRUE,
-                  "ssl-strict", FALSE,
+                  "ssl-strict", TRUE,
                   NULL);
     #elif defined (HAVE_LIBSOUP_2_29_91)
     const gchar* certificate_files[] =
@@ -191,7 +191,7 @@
         {
             g_object_set (session,
                 "ssl-ca-file", certificate_files[i],
-                "ssl-strict", FALSE,
+                "ssl-strict", TRUE,
                 NULL);
             break;
         }
@@ -206,7 +206,7 @@
     gchar* certificate_file = midori_paths_get_res_filename ("ca-bundle.crt");
     g_object_set (session,
                   "ssl-ca-file", certificate_file,
-                  "ssl-strict", FALSE,
+                  "ssl-strict", TRUE,
                   NULL);
     g_free (certificate_file);
     #endif
